src/transform.ts

```diff
@@ -54,27 +51,27 @@ export function getTransactionFromBlockPayout(block: algosdk.modelsv2.BlockRespo
  * @param blockResponse An Algorand block
  * @returns An array of processed transactions from the block
  */
-export function getBlockTransactions(blockResponse: algosdk.modelsv2.BlockResponse): TransactionInBlock[] {
+export function getBlockTransactions(blockResponse: BlockResponse): TransactionInBlock[] {
   const block = blockResponse.block
 
   let roundOffset = 0
   const getRoundOffset = () => roundOffset++
 
   const txns = (block.payset ?? []).flatMap((signedTransactionInBlock) => {
     // Assume that Consensus.RequireGenesisHash is true, we always copy the genesis hash from the block to the transaction
-    const genesisHash = Buffer.from(blockResponse.block.header.genesisHash)
-    const genesisId = signedTransactionInBlock.hasGenesisID ? blockResponse.block.header.genesisID : undefined
+    const genesisHash = block.header.genesisHash ? Buffer.from(block.header.genesisHash) : undefined
+    const genesisId = signedTransactionInBlock.hasGenesisId ? block.header.genesisId : undefined
 
     const rootTransactionData = extractTransactionDataFromSignedTxnInBlock(signedTransactionInBlock.signedTxn, genesisHash, genesisId)
 
     const rootTransaction = {
       signedTxnWithAD: signedTransactionInBlock.signedTxn,
-      transactionId: rootTransactionData.transaction.txID(),
+      transactionId: rootTransactionData.transaction.txId(),
       intraRoundOffset: getRoundOffset(),
-      roundNumber: block.header.round,
+      roundNumber: block.header.round!,
```

Comment:
Definitely feels like this shouldn't be optional. We might need to make some tweaks to the block model.